[Mind-23] 매칭 목록 조회 기능 추가

TaskerJang opened the pull request:
# Pull Request

## 💡 PR 요약
매칭 목록 조회 기능을 추가했습니다. 사용자가 본인의 매칭 목록을 조회할 수 있는 기능을 구현했습니다.

## 🔍 주요 변경사항
- 매칭 엔티티에 내 매칭 목록 조회 기능 추가
- 매칭 저장소(Repository)에 findByUserId 메서드 구현
- 내 매칭 목록 조회 API 컨트롤러 엔드포인트 추가
- 매칭 서비스에 getMyMatchings 메서드 구현
- 매칭 목록 조회 기능 테스트 코드 작성

## 🔗 연관된 이슈
#23 내 매칭 목록 조회 기능 구현

## ✅ 체크리스트
- [x] 테스트 코드를 작성하였나요?
- [x] 관련 문서를 업데이트하였나요?
- [ ] Breaking Change가 있나요?
- [x] 코드 포맷팅과 린트 검사를 완료하였나요?

## 🙏 리뷰어 참고사항


## 📋 추가 컨텍스트

Commit:
[feat]: 내 매칭 목록 조회 기능 구현 (#23)
- 사용자별 매칭 목록 조회 API 구현
- MatchingRepository에 findByUserId 메서드 추가
- MatchingService에 매칭 목록 조회 메서드 구현

Commit:
[test]: 내 매칭 목록 조회 기능 테스트 코드 추가 (#23)
- 매칭 목록 조회 성공 케이스 테스트
- 매칭이 없는 경우 빈 목록 반환 테스트
- 다른 사용자의 매칭이 포함되지 않는지 테스트

hwan2-99 commented on src/main/java/com/mindgoal/domain/matching/repository/MatchingRepository.java:
```diff
@@ -1 +1,22 @@
+     * @param userId 사용자 ID
+     * @return 매칭 목록
+     */
+    List<Matching> findByUserId(Long userId);
```
메서드명이 잘못된것 같습니다! 아마 JPA의 명명규칙때문에 findBy~~는 Optional로 하나만 불러오거나 없을경우 empty를 반환하기떄문에 findAllByUserId로 수정하시면 좋을것 같습니다!

hwan2-99 commented on src/main/java/com/mindgoal/domain/matching/service/MatchingService.java:
```diff
@@ -1 +1,87 @@
+
+        return matchings.stream()
+                .map(MatchingResponseDto::from)
+                .collect(Collectors.toList());
```
저희는 자바 17버전이기때문에 
```java
return matchings.stream()
                .map(MatchingResponseDto::from)
                .toList();
```
로 사용할수 있을것 같습니다! :)

hwan2-99 reviewed (commented):
고생하셨습니다!!

Commit:
[refactor]: 매칭 관련 클래스 최신 Java 기능 활용

- MatchingRepository의 메서드명 findByUserId → findAllByUserId로 변경
- MatchingService에서 Stream API의 Java 17 toList() 메서드 적용

Commit:
conflict resolve
